### PR TITLE
Workaround broken Shiboken 6 getattr

### PR DESCRIPTION
**Pull request by Jookia**
This adds a workaround for #14101. Rather than crashing it displays a regular list of attributes.

### CODE REVIEW

File: src/Gui/CallTips.cpp [hunk 1]
```diff
@@ -752,4 +732,56 @@ QString CallTipsList::stripWhiteSpace(const QString& str) const
...
+
+    if (name == QLatin1String("__doc__") && attr.isString()) {
+        Py::Object help = attr;
+        if (help.isString()) {
+            Py::String doc(help);
+            QString longdoc = QString::fromUtf8(doc.as_string().c_str());
+            int pos = longdoc.indexOf(QLatin1Char('\n'));
+            pos = qMin(pos, 70);
+            if (pos < 0)
+                pos = qMin(longdoc.length(), 70);
```
Comment (chennes) [inline src/Gui/CallTips.cpp:765]: Could you please add parenthesis around this clause (and the one below)?

**Comment by Jookia** [comment]
Did you want me to reformat the code when moving it?

**Comment by chennes** [comment]
Basically, if a side-by-side diff *isn't* side by side (as is the case here) then reformatting is fair game. I don't really care about the tiny things, but these single-line conditionals are an infamous source of bugs. IMO we should clean them up when we are touching the code already.

**Comment by Jookia** [comment]
Is there a clang-format for this?

**Comment by chennes** [comment]
You can use the project-wide .clang-format file, just be careful that you only apply it to lines you are already touching for one reason or another. The Gui subdirectory is not yet under our format-enforcing pre-commit action.

**Comment by Jookia** [comment]
I decided to just apply a cleanup to the CallTips files while I'm here. I know that's slightly out of scope but it seems like a good time to do it.

If this is unwanted I'll go through and only do the changes in the code I've changed.

**Comment by chennes** [comment]
I don't mind in this case since it is in a separate commit that we can add to .git-blame-ignore-revs.

**Comment by Jookia** [comment]
Did you want me to add it to .git-blame-ignore-revs too? :)

**Comment by Jookia** [comment]
Looks like this is breaking the Windows CI build:

```
CallTips.cpp.obj : error LNK2019: unresolved external symbol "bool __cdecl Py::operator>=(class Py::Long const &,int)" (??PPy@@YA_NAEBVLong@0@H@Z) referenced in function "bool __cdecl shibokenMayCrash(void)" (?shibokenMayCrash@@YA_NXZ)
```

I'll look in to this when I can.

**Comment by chennes** [comment]
> Did you want me to add it to .git-blame-ignore-revs too? :)
> […](#)
> On Sun, Jun 23, 2024 at 07:57:03PM -0700, Chris Hennes wrote: I don't mind in this case since it is in a separate commit that we can add to .git-blame-ignore-revs. -- Reply to this email directly or view it on GitHub: [#14864 (comment)](https://github.com/FreeCAD/FreeCAD/pull/14864#issuecomment-2185492647) You are receiving this because you authored the thread. Message ID: ***@***.***>

Wait until it's merged, the hash will change ever time the PR is rebased.

**Comment by Jookia** [comment]
Oh, that makes sense.

**Comment by xtemp09** [comment]
I can no longer reproduce the crash reported with these commits. Note that the version of _Python_ is different and I compile _FreeCAD_ with _Boost.Python_ instead of _PySide_.



```ruby
OS: Ubuntu 22.04.4 LTS (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37843 (Git)
Build type: Debug
Branch: main
Hash: 932c73f0de2d1e9ba3f60ae14d199cfe77255c3a
Python 3.10.12, Qt 6.7.1, Coin 4.0.2, Vtk 9.1.0, OCC 7.5.1
Locale: English/United States (en_US)
```

**Comment by Jookia** [comment]
That's great news, thanks for the test.

**Comment by pieterhijma** [comment]
My system has updated in the meantime (Qt 6.7.1 instead of Qt 6.7.0) compared to #14101, but I can confirm that the issue is gone.

```
OS: Arch Linux
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37844 +3 (Git)
Build type: DEBUG
Branch: RFC_shiboken6fix
Hash: 3149af2ee66208608e28cb17ac3e31acce8f19be
Python 3.12.3, Qt 6.7.1, Coin 4.0.2, Vtk , OCC 7.7.2
Locale: English/United States (en_US)
```

**Comment by Jookia** [comment]
I'm converting this to a draft as after talking in the Qt bug report, the expectation is that we should have a stack frame when using PySide6 in the first place. So this may be the wrong fix.

**Comment by xtemp09** [comment]
@Jookia, what back trace have you got when you reproduced the crash? Here is mine:

<details><summary>Backtrace </summary>


```cpp
Process 1720 stopped
* thread #1, name = 'FreeCAD', stop reason = signal SIGSEGV: address not mapped to object (fault address: 0x8)
    frame #0: 0x00007ffff532ff06 libpython3.10.so.1.0`PyObject_GetAttr + 22
libpython3.10.so.1.0`PyObject_GetAttr:
->  0x7ffff532ff06 <+22>: movq   0x8(%rdi), %rbx
    0x7ffff532ff0a <+26>: testb  $0x10, 0xab(%rax)
    0x7ffff532ff11 <+33>: je     0x7ffff532ffe8 ; <+248>
    0x7ffff532ff17 <+39>: movq   0x90(%rbx), %rax
(lldb) bt
* thread #1, name = 'FreeCAD', stop reason = signal SIGSEGV: address not mapped to object (fault address: 0x8)
  * frame #0: 0x00007ffff532ff06 libpython3.10.so.1.0`PyObject_GetAttr + 22
    frame #1: 0x00007fffe4759700 libshiboken6.abi3.so.6.7`___lldb_unnamed_symbol1450 + 128
    frame #2: 0x00007fffe4766025 libshiboken6.abi3.so.6.7`mangled_type_getattro + 1077
    frame #3: 0x00007ffff532ff2b libpython3.10.so.1.0`PyObject_GetAttr + 59
    frame #4: 0x00007ffff53300c3 libpython3.10.so.1.0`PyObject_GetAttrString + 67
    frame #5: 0x00007ffff6f12e27 libFreeCADGui.so`Py::Object::getAttr(this=<unavailable>, s=<unavailable>) const at Objects.hxx:274:94
    frame #6: 0x00007ffff71f9bae libFreeCADGui.so`Gui::CallTipsList::extractTipsFromObject(this=0x0000555555c33ef0, obj=0x00007fffffffc2d0, list=0x00007fffffffc300, tips=0x00007fffffffc410) const at CallTips.cpp:360:47
    frame #7: 0x00007ffff71faed9 libFreeCADGui.so`Gui::CallTipsList::extractTips(this=0x0000555555c33ef0, context=0x00007fffffffc420) const at CallTips.cpp:333:30
    frame #8: 0x00007ffff71fbb74 libFreeCADGui.so`Gui::CallTipsList::showTips(this=0x0000555555c33ef0, line=0x00007fffffffc560) at CallTips.cpp:485:54
    frame #9: 0x00007ffff72085be libFreeCADGui.so`Gui::PythonConsole::keyPressEvent(this=0x0000555555d510a0, e=<unavailable>) at PythonConsole.cpp:675:44
    frame #10: 0x00007ffff4be7fa0 libQt6Widgets.so.6`QWidget::event(QEvent*) + 4144
    frame #11: 0x00007ffff4c779b6 libQt6Widgets.so.6`QFrame::event(QEvent*) + 54
    frame #12: 0x00007ffff4b8b345 libQt6Widgets.so.6`QApplicationPrivate::notify_helper(QObject*, QEvent*) + 133
    frame #13: 0x00007ffff4b94b2d libQt6Widgets.so.6`QApplication::notify(QObject*, QEvent*) + 3501
    frame #14: 0x00007ffff6fbee58 libFreeCADGui.so`Gui::GUIApplication::notify(this=0x00007fffffffd6b0, receiver=0x0000555555d510a0, event=0x00007fffffffcec0) at GuiApplication.cpp:83:40
    frame #15: 0x00007ffff377b398 libQt6Core.so.6`QCoreApplication::notifyInternal2(QObject*, QEvent*) + 344
    frame #16: 0x00007ffff4bfc540 libQt6Widgets.so.6`QWidgetWindow::event(QEvent*) + 512
    frame #17: 0x00007ffff4b8b345 libQt6Widgets.so.6`QApplicationPrivate::notify_helper(QObject*, QEvent*) + 133
    frame #18: 0x00007ffff6fbee58 libFreeCADGui.so`Gui::GUIApplication::notify(this=0x00007fffffffd6b0, receiver=0x0000555559dec170, event=0x00007fffffffcec0) at GuiApplication.cpp:83:40
    frame #19: 0x00007ffff377b398 libQt6Core.so.6`QCoreApplication::notifyInternal2(QObject*, QEvent*) + 344
    frame #20: 0x00007ffff3fcc73b libQt6Gui.so.6`QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) + 379
    frame #21: 0x00007ffff40300fc libQt6Gui.so.6`QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 172
    frame #22: 0x00007fffedb9ec4e libQt6XcbQpa.so.6`xcbSourceDispatch(_GSource*, int (*)(void*), void*) + 30
    frame #23: 0x00007ffff231bd3b libglib-2.0.so.0`g_main_context_dispatch + 619
    frame #24: 0x00007ffff23712b8 libglib-2.0.so.0`___lldb_unnamed_symbol2709 + 488
    frame #25: 0x00007ffff23193e3 libglib-2.0.so.0`g_main_context_iteration + 51
    frame #26: 0x00007ffff3a53560 libQt6Core.so.6`QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 112
    frame #27: 0x00007ffff3787b7b libQt6Core.so.6`QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 427
    frame #28: 0x00007ffff3783ffa libQt6Core.so.6`QCoreApplication::exec() + 154
    frame #29: 0x00007ffff6f0b7ed libFreeCADGui.so`(anonymous namespace)::tryRunEventLoop(mainApp=0x00007fffffffd6b0) at Application.cpp:2044:23
    frame #30: 0x00007ffff6f0badd libFreeCADGui.so`(anonymous namespace)::runEventLoop(mainApp=0x00007fffffffd6b0) at Application.cpp:2064:24
    frame #31: 0x00007ffff6f0ef04 libFreeCADGui.so`Gui::Application::runApplication() at Application.cpp:2141:17
    frame #32: 0x000055555556a3d7 FreeCAD`main(argc=<unavailable>, argv=<unavailable>) at MainGui.cpp:300:45
    frame #33: 0x00007ffff2e29d90 libc.so.6`__libc_start_call_main(main=(FreeCAD`main at MainGui.cpp:107:1), argc=1, argv=0x00007fffffffdd48) at libc_start_call_main.h:58:16
    frame #34: 0x00007ffff2e29e40 libc.so.6`__libc_start_main_impl(main=(FreeCAD`main at MainGui.cpp:107:1), argc=1, argv=0x00007fffffffdd48, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffdd38) at libc-start.c:392:3
    frame #35: 0x0000555555569c05 FreeCAD`_start + 37
(lldb)
```


</details>

**Comment by Jookia** [comment]
Yes, that's the trace I got.

**Comment by Jookia** [comment]
I'm obsoleting this in favor of #15236 which provides a much better fix.

**Pull request closed without merging**